Detections/AzureDiagnostics/KeyvaultMassSecretRetrieval.yaml

```diff
@@ -25,15 +25,25 @@ query: |
   ["SecretGet", "KeyGet", "VaultGet"]);
   AzureDiagnostics
   | where TimeGenerated between (ago(starttime)..ago(endtime))
+  | extend ResultType = columnifexists("ResultType", "None"), identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g = columnifexists("identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g", "None")
+  | where ResultType != "None" and isnotempty(ResultType)
+  | where identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g != "None" and isnotempty(identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g)
   | where ResourceType == "VAULTS" and ResultType == "Success"
   | where OperationName in (OperationList) 
   | summarize count() by identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g, OperationName
   | where count_ > AlertTriggerThreshold  
   | join (
   AzureDiagnostics
   | where TimeGenerated between (ago(starttime)..ago(endtime))
-  | where ResourceType == "VAULTS" and ResultType == "Success"
+  | extend ResultType = columnifexists("ResultType", "NoResultType")
+  | extend ResultType = columnifexists("ResultType", "None"), identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g = columnifexists("identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g", "None")
+  | extend id_s = columnifexists("id_s", "None"), CallerIPAddress = columnifexists("CallerIPAddress", "None"), clientInfo_s = columnifexists("clientInfo_s", "None")
+  | where ResultType != "None" and isnotempty(ResultType)
+  | where identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g != "None" and isnotempty(identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g)
+  | where id_s != "None" and isnotempty(id_s)
+  | where CallerIPAddress != "None" and isnotempty(CallerIPAddress)
+  | where clientInfo_s != "None" and isnotempty(clientInfo_s)
   | where OperationName in (OperationList)   
   ) on identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g 
   | project TimeGenerated, ResourceType, ResultType, Resource, OperationName, id_s, CallerIPAddress, identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g, requestUri_s, clientInfo_s
```

Comment:
requestUri_s is not a property that is defined.

Reply:
added fix.